packages/upload/Upload.js

```diff
@@ -45,6 +45,7 @@ class Upload extends Component {
 
   setFiles = files => {
     let selectedFiles = [];
+    const id = uuid();
```

Comment:
this change guarantees that every file passed into `setFiles` will be appended with the same id.

Reply:
you are correct, this might not be best solution. I'll look into it